classes/ActionScheduler_Abstract_QueueRunner.php

```diff
@@ -79,6 +92,115 @@ public function get_allowed_concurrent_batches() {
 		return apply_filters( 'action_scheduler_queue_runner_concurrent_batches', 5 );
 	}
 
+	/**
+	 * Get the maximum number of seconds a batch can run for.
+	 *
+	 * @return int The number of seconds.
+	 */
+	protected function get_maximum_execution_time() {
+
+		// There are known hosts with a strict 60 second execution time.
+		if ( defined( 'WPENGINE_ACCOUNT' ) || defined( 'PANTHEON_ENVIRONMENT' ) ) {
+			$maximum_execution_time = 60;
+		} elseif ( false !== strpos( getenv( 'HOSTNAME' ), '.siteground.' ) ) {
+			$maximum_execution_time = 120;
+		} else {
+			$maximum_execution_time = ini_get( 'max_execution_time' );
+		}
+
+		return absint( apply_filters( 'action_scheduler_maximum_execution_time', $maximum_execution_time ) );
+	}
+
+	/**
+	 * Get the number of seconds a batch has run for.
+	 *
+	 * @return int The number of seconds.
+	 */
+	protected function get_execution_time() {
+		$execution_time = microtime( true ) - $this->created_time;
+
+		// Get the CPU time if the hosting environment uses it rather than wall-clock time to calculate a process's execution time.
+		if ( function_exists( 'getrusage' ) && apply_filters( 'action_scheduler_use_cpu_execution_time', defined( 'PANTHEON_ENVIRONMENT' ) ) ) {
+			$resource_usages = getrusage();
+
+			if ( isset( $resource_usages['ru_stime.tv_usec'], $resource_usages['ru_stime.tv_usec'] ) ) {
+				$execution_time = $resource_usages['ru_stime.tv_sec'] + ( $resource_usages['ru_stime.tv_usec'] / 1000000 );
+			}
+		}
+
+		return $execution_time;
+	}
+
+	/**
+	 * Check if the host's max execution time is (likely) to be exceeded if processing more actions.
+	 *
+	 * @param int $processed_actions The number of actions processed so far - used to determine the likelihood of exceeding the time limit if processing another action
+	 * @return bool
+	 */
+	protected function time_likely_to_be_exceeded( $processed_actions ) {
+
+		$execution_time      = $this->get_execution_time();
+		$max_execution_time  = $this->get_maximum_execution_time();
+		$time_per_action     = $execution_time / $processed_actions;
+		$estimated_time      = $execution_time + ( $time_per_action * 2 );
+		$likely_to_be_exceed = ( $estimated_time > $this->get_maximum_execution_time() ) ? true : false;
```

Comment:
You can drop `? true : false` from the end of this line. This is redundant since the `>` operator (and other comparison operators) will return a boolean value. You can also omit the parentheses around the condition.

One minor thing here is the name of the variable. I'd suggest either `$likely_to_exceed` or `$likely_to_be_exceeded`.